portal/src/ThemeConfigurationWidget.tsx

```diff
@@ -212,6 +212,49 @@ const ThemeConfigurationWidget: React.FC<ThemeConfigurationWidgetProps> =
       [onChangeDarkModeEnabled]
     );
 
+    const [customTheme, setCustomTheme] = useState<
+      Omit<LightTheme | DarkTheme, "isDarkTheme">
+    >(() => {
+      if (isDarkMode) {
+        if (darkTheme) {
+          const { isDarkTheme, ...colors } = darkTheme;
+          return colors;
+        }
+        const { isDarkTheme, ...colors } = DEFAULT_DARK_THEME;
+        return colors;
+      }
+      if (lightTheme) {
+        const { isDarkTheme, ...colors } = lightTheme;
+        return colors;
+      }
+      const { isDarkTheme, ...colors } = DEFAULT_LIGHT_THEME;
+      return colors;
+    });
+
+    const onPickerChangePrimaryColor = useCallback(
+      (color: string) => {
+        setCustomTheme({ ...customTheme, primaryColor: color });
```

Comment:
setState can accept an update function. By using the update function, we can avoid depending on the current state.